reconcile/test/test_container_registry_mirror/test_engine.py

```
@@ -0,0 +1,571 @@
+            source_creds="src_user:src_pass",
+            destination_url="quay.io/org/image",
+            destination_creds="dst_user:dst_pass",
+        )
+
+        source, dest = engine._build_images(spec)
+
+        # Image.__str__ prepends "docker://" and appends ":latest".
+        assert "docker.io/upstream/image" in str(source)
+        assert "quay.io/org/image" in str(dest)
+        assert source.username == "src_user"
+        assert source.password == "src_pass"
```

Ruff (0.15.21)

[error] Possible hardcoded password assigned to: "password"

(S105)